test/watcher_test.js

```diff
@@ -139,6 +162,65 @@ describe('Watcher', function() {
       expect(builderBuild).to.have.been.called;
     });
 
+    it('on change, rebuild is invoked and cancel is invoked from the build', function() {
+      const pluginsCalled = {};
+
+      const waitForPlugin1 = defer();
+      const waitForPlugin2 = defer();
+
+      class Plugin1 extends Plugin {
+        constructor(inputNodes, options) {
+          super(inputNodes || [], options);
+        }
+
+        build() {
+          pluginsCalled['plugin1'] = pluginsCalled['plugin1'] ? pluginsCalled['plugin1'] + 1 : 1;
```

Comment:
Once migrated to an array of "build events" (see prior comment), this can become:

```suggestion
    events.push('plugin1 build');
```